app/code/Magento/Catalog/Test/Unit/Model/ProductRepositoryTest.php

```diff
@@ -345,6 +346,66 @@ function ($value) {
         $this->objectManager->setBackwardCompatibleProperty($this->model, 'mediaProcessor', $mediaProcessor);
     }
 
+    /**
+     * Test save product with store id 0
+     *
+     * @param array $productData
+     * @return void
+     * @dataProvider getProductData
+     */
+    public function testSaveForAllStoreViewScope(array $productData): void
+    {
+        $this->productFactory->method('create')->willReturn($this->product);
+        $this->product->method('getSku')->willReturn($productData['sku']);
+        $this->extensibleDataObjectConverter
+            ->expects($this->once())
+            ->method('toNestedArray')
+            ->willReturn($productData);
+        $this->resourceModel->method('getIdBySku')->willReturn(100);
+        $this->resourceModel->expects($this->once())->method('validate')->willReturn(true);
+        $this->product->expects($this->at(14))->method('setData')->with('store_id', $productData['store_id']);
```

Comment:
This looks like too tight coupling to implementation. Try to replace in original class `setData` into `setStoreId(...)`.

Reply:
https://github.com/magento/magento2/blob/06a0dd0d5f7ea3e389dc5fa718400866ab4c1eef/app/code/Magento/Catalog/Model/ProductRepository.php#L380-L382
StoreId is one item of ProductData array so setting it separately is redundant.

Reply:
So why it's actually `foreach` instead of just `$product->setData($productData)` o.O

Reply:
In the case, if we pass an array in $key argument setData will overwrite all the data in the object.

Reply:
I see :-) OK.